logstash-core/src/main/java/org/logstash/secret/store/SecretStoreFactory.java

```diff
@@ -27,29 +25,73 @@ private SecretStoreFactory() {
+    }
+
     @SuppressWarnings({"unchecked", "JavaReflectionMemberAccess"})
-    static public SecretStore loadSecretStore(SecureConfig secureConfig) {
-        //cheap SPI, if we ever support more then one implementation we should expose it as a setting and push the class name here via the secureConfig
-        String className = System.getProperty("org.logstash.secret.store.SecretStore", "org.logstash.secret.store.backend.JavaKeyStore");
+    private static SecretStore doIt(MODE mode, SecureConfig secureConfig) {
+        char[] configuredClassName = secureConfig.getPlainText("keystore.classname");
+        String className = configuredClassName != null ? new String(configuredClassName) : "org.logstash.secret.store.backend.JavaKeyStore";
         try {
-            LOGGER.debug("Attempting to loadSecretStore secret store with implementation: {}", className);
+            LOGGER.debug("Attempting to {} or secret store with implementation: {}", mode.name().toLowerCase(), className);
             Class<? extends SecretStore> implementation = (Class<? extends SecretStore>) Class.forName(className);
-            Constructor<? extends SecretStore> constructor = implementation.getConstructor(SecureConfig.class);
+
             addSecretStoreAccess(secureConfig);
-            return constructor.newInstance(secureConfig);
-        } catch (InstantiationException | IllegalAccessException | ClassNotFoundException | NoSuchMethodException e) {
+
+            if (MODE.LOAD.equals(mode)) {
+                return implementation.newInstance().load(secureConfig);
+            } else if (MODE.CREATE.equals(mode)) {
+                return implementation.newInstance().create(secureConfig);
+            } else if (MODE.DELETE.equals(mode)) {
+                implementation.newInstance().delete(secureConfig);
+                return null;
+            } else if (MODE.EXISTS.equals(mode)) {
+                return implementation.newInstance();
+            } else {
+                throw new IllegalStateException("missing mode. This is bug in Logstash.");
+            }
+        } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
             throw new SecretStoreException.ImplementationNotFoundException(
-                    String.format("Could not loadSecretStore class %s, please ensure it is on the Java classpath, implements org.logstash.secret.store.SecretStore, and has a 1 " +
-                            "argument constructor that accepts a org.logstash.secret.store.SecureConfig", className), e);
-        } catch (InvocationTargetException ite) {
-            //this is thrown if an exception is thrown from the constructor
-            throw new SecretStoreException.LoadException("Unable to load Logstash secret store", ite);
+                    String.format("Could not %s class %s, please ensure it is on the Java classpath, implements org.logstash.secret.store.SecretStore, and has a zero " +
```

Comment:
[commentary, no change required}

I am feeling that this message is not necessarily going to be useful to any operator/users -- 

"please ensure it is on the Java classpath"

I am feeling that if someone hits this exception, this is a bug, not a misconfiguration? If it is a misconfiguration, I think we would ideally provide a more actionable error message/

Reply:
Agreed, I changed it to

```java
String.format("Could not %s class %s, please validate the `keystore.classname` is configured correctly and that the class can be loaded by Logstash ", mode
                                    .name().toLowerCase(), className), e);
```